From: "Martin T. H. Sandsmark" <martin.sandsmark@kde.org>
Date: Wed, 25 Jan 2017 22:25:48 +0100
Subject: Fix opening of new tabs with multiple processes

Make sure we re-use the process if the user wants a new tab.

BUG: 373440
REVIEW: 129886
---
 src/main.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index fb53560e..db3b83bf 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -121,8 +121,8 @@ extern "C" int Q_DECL_EXPORT kdemain(int argc, char* argv[])
     parser->process(args);
     about.processCommandLine(parser.data());
 
-    // Enable user to force multiple instances
-    if (!Konsole::KonsoleSettings::useSingleInstance()) {
+    // Enable user to force multiple instances, unless a new tab is requested
+    if (!Konsole::KonsoleSettings::useSingleInstance() && !parser->isSet(QStringLiteral("new-tab"))) {
         startupOption = KDBusService::Multiple;
     }
 
